app/components/swap/swap.tsx

```diff
@@ -0,0 +1,293 @@
+
+  const updateRate = () => {
+    const fromPrice = tokenPrices[fromToken];
+    const toPrice = tokenPrices[toToken];
+    const newRate = toPrice / fromPrice;
+    setRate(newRate);
+  };
+
+  const numberRegex = /^[0-9]*[.,]?[0-9]*$/;
+
+  const handleConnectWallet = async () => {
+    if (connectors.length > 0) {
+      try {
+        await connect({ connector: connectors[0] });
+      } catch (error) {
+        console.error("Failed to connect wallet:", error);
+        setError("Failed to connect wallet. Please try again.");
+      }
+    }
+  };
+
+  const handleSwap = async () => {
+    if (!address) return;
+
+    setIsLoading(true);
+    setError("");
+
+    // Simulate a delay for the swap process
+    setTimeout(() => {
+      try {
+        const swappedAmount = parseFloat(amount) * rate;
+        console.log(
+          `Swapped ${amount} ${fromToken} for ${swappedAmount.toFixed(6)} ${toToken}`
+        );
+        setAmount("");
+        setEquivalent("0");
+        setError("");
+      } catch (error) {
+        console.error("Swap failed:", error);
+        setError("Swap failed. Please try again.");
+      } finally {
+        setIsLoading(false);
+      }
+    }, 2000);
+  };
+
+  const handleTokenSwap = () => {
+    const temp = fromToken;
+    setFromToken(toToken);
+    setToToken(temp);
+  };
+
+  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    if (numberRegex.test(value)) {
+      setAmount(value);
+    }
+  };
+
+  return (
+    <div
+      className={`mt-12 flex h-[371.67px] w-full cursor-pointer flex-col items-center gap-2 rounded-full md:h-[636px] md:w-[594px] md:gap-6 ${theme === "dark" ? "bg-light-linear-gradient" : "bg-dark-linear-gradient"} rounded-xl p-[18px] md:p-[32px]`}
+    >
+      <div
+        className={`flex w-full items-center justify-between ${theme === "dark" ? "text-black" : "text-white"}`}
+      >
+        <p className="text-[15px] font-[700] md:text-[25px]">Swap</p>
+        <Image
+          src={theme === "dark" ? "/icons/close.svg" : "/icons/close-white.svg"}
+          alt="Close"
+          width={40}
+          height={40}
+          className="h-[20px] w-[20px] md:h-[40px] md:w-[40px]"
+        />
+      </div>
+
+      <div className="relative flex w-full flex-col items-center">
+        <div className="mb-4 flex w-full flex-col">
+          <label
+            className={`mb-2 text-[9.97px] md:text-[16px] ${theme === "dark" ? "text-black" : "text-white"}`}
+          >
+            From
+          </label>
+          <div
+            className={`flex h-[78.49px] flex-col items-center rounded-[8px] px-[10px] py-[10px] md:h-[134px] md:px-[24px] md:py-[30px] ${theme === "dark" ? "border border-[#dadada] bg-[#f5f5f5]" : "border border-grey-700 bg-grey-600"}`}
+          >
+            <div className="flex w-full items-center justify-between">
+              <div className="flex flex-col items-start">
+                <input
```

Comment:
0 should be a placeholder not the value, so that it doesn't append to the actual input

Reply:
fixed